# joint names
string[] names

# joint IDs (if empty, use order of name array)
int8[] ids